neo/SmartContract/ApplicationEngine.cs

```diff
@@ -496,98 +496,49 @@ protected virtual long GetPrice(OpCode nextInstruction)
             }
         }
 
-        protected virtual long GetPriceForSysCall()
+        public virtual long GetPriceForSysCall()
```

Comment:
To test the prices for syscalls, you don't need to mark `GetPriceForSysCall` as `public`. Just run the script and check the property `GasConsumed`.

Reply:
I think it's good that it's public, sometimes we need a lot of solutions for easy jobs

Reply:
But I think it should be excellent if the cost of the gas is handled by 'InteropService', is more centralized, like I did in NeoSharp

https://github.com/CityOfZion/neo-sharp/blob/development/src/NeoSharp.VM/Types/InteropService.cs#L59-L83

It's easy to verify before execution.

https://github.com/CityOfZion/neo-sharp/blob/development/src/NeoSharp.VM/Types/InteropService.cs#L59-L83

Reply:
Ok, it will go back to protected, because that's not the point of this PR... On the other hand, I will propose an amazing testing feature using Reflection, that I just learned today :)